File: cloud_testing/cloud_api_poc/src/main/java/cloud_api_poc/GcStorage.java
package cloud_api_poc;

import static cloud_api_poc.Config.bucketGcsPath;
import static cloud_api_poc.Constants.GCS_PREFIX;
import static cloud_api_poc.Utils.fatalError;

import com.google.cloud.storage.BlobInfo;
import com.google.cloud.storage.Storage;
import com.google.cloud.storage.StorageOptions;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public abstract class GcStorage {
  private GcStorage() {}

  private static final Storage storage = StorageOptions.newBuilder().build().getService();

  public static String uploadApk(Path apk) {
    String apkFileName = apk.getFileName().toString();
    String gcsApkPath = GCS_PREFIX + Paths.get(bucketGcsPath, apkFileName).toString();

    // todo: check if bucketGcsPath exists. create if not.
    // 404 Not Found error when bucketGcsPath does not exist
    BlobInfo apkBlob = BlobInfo.newBuilder(bucketGcsPath, apkFileName).build();

    try {
      storage.create(apkBlob, Files.readAllBytes(apk));
    } catch (Exception e) {
      fatalError(e);
    }

    return gcsApkPath;
  }
}

File: cloud_testing/kotlin_poc/src/main/kotlin/GcStorage.kt
import com.google.cloud.storage.BlobInfo
import com.google.cloud.storage.StorageOptions

import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths

import GlobalConfig.bucketGcsPath
import Constants.GCS_PREFIX
import Utils.fatalError

object GcStorage {

    private val storage = StorageOptions.newBuilder().build().service

    fun uploadApk(apk: Path): String {
        val apkFileName = apk.fileName.toString()
        val gcsApkPath = GCS_PREFIX + Paths.get(bucketGcsPath, apkFileName).toString()

        // todo: check if bucketGcsPath exists. create if not.
        // 404 Not Found error when bucketGcsPath does not exist
        val apkBlob = BlobInfo.newBuilder(bucketGcsPath, apkFileName).build()

        try {
            storage.create(apkBlob, Files.readAllBytes(apk))
        } catch (e: Exception) {
            fatalError(e)
        }

        return gcsApkPath
    }
}
